### DISCUSSION TITLE
Everything is inverted

### DISCUSSION THREAD

**Discussion by sergedc** [powerpaul17/nc_money#18] [General]
Hi, 
Nice tool, but what am I not getting: if I get 100 of income from my work, my bank account (asset) has 100. 
Currently when you put 100 of income, Assets are going to -100

**Comment by powerpaul17** [comment]
Hello, what you observed is due to the nature of double-entry accounting. If it goes into your bank account it has to come from somewhere, in your case the income account. So mathematically this is going to be a negative value. As far as I know accountants use two columns for keeping track and in income accounts it is labeled differently but I'm not an accountant unfortunately.. :sweat_smile: 

But I'm aware of this issue and that it seems not intuitive (even if mathematically correct) and I am thinking about simply inversing the values of the income accounts but I want to investigate the possible solutions more before I start implementing something..

Hope I made it clear somehow.. :sweat_smile:

**Comment by sergedc** [comment]
I am an accountant and double entry accouting means that each entry is reflected 2 times.An income can either be reflected as an increase in asset or decrease in liabilities. An expense can either be reflected as a decreade in asset or an increase in liability. Double entry accounting is not "it has to come from somewhere"  ---- On Thu, 06 Oct 2022 03:46:25 -0400  ***@***.***  wrote ----

Hello, what you observed is due to the nature of double-entry accounting. If it goes into your bank account it has to come from somewhere, in your case the income account. So mathematically this is going to be a negative value. As far as I know accountants use two columns for keeping track and in income accounts it is labeled differently but I'm not an accountant unfortunately.. 😅

But I'm aware of this issue and that it seems not intuitive (even if mathematically correct) and I am thinking about simply inversing the values of the income accounts but I want to investigate the possible solutions more before I start implementing something..

Hope I made it clear somehow.. 😅

**Comment by powerpaul17** [comment]
It is reflected 2 times. One time you have +100 on your asset account. And on the other side you have -100 on your income account so the net is zero. Then it's a balanced transaction (sorry if I don't have the correct wording as I said, I'm not an accountant :wink: ).

This shows more of what I mean: [https://www.quora.com/Why-do-income-accounts-have-negative-signs-in-ledgers?share=1](https://www.quora.com/Why-do-income-accounts-have-negative-signs-in-ledgers?share=1)

**Comment by powerpaul17** [comment]
Or in other words, I'm currently only depicting the __debits__ column of an account table (due to space constraints) so it can become negative.
